File: account-service/src/main/java/com/banvien/training/controller/UserController.java
package com.banvien.training.controller;

import com.banvien.training.dto.request.PaymentRequest;
import com.banvien.training.dto.response.StatusPaymentResponse;
import com.banvien.training.dto.response.UserBalanceResponse;
import com.banvien.training.dto.response.UserResponse;
import com.banvien.training.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("users")
@RequiredArgsConstructor
public class UserController {

    private final UserService userService;

    @GetMapping("/{userId}")
    public UserResponse getUser(@PathVariable Long userId) {
        return userService.getUser(userId);
    }

    @GetMapping("/{userId}/balance")
    public UserBalanceResponse getUserBalance(@PathVariable Long userId) {
        return userService.getUserBalance(userId);
    }

    @PostMapping("/{userId}/deduct-money")
    public StatusPaymentResponse deductMoney(@RequestBody PaymentRequest request, @PathVariable Long userId) {
        return userService.deductMoney(userId, request.getAmount());
    }

    @PostMapping("/{userId}/refund-money")
    public StatusPaymentResponse refundMoney(@RequestBody PaymentRequest request, @PathVariable Long userId) {
        return userService.refundMoney(userId, request.getAmount());
    }

}

File: gateway-service/app/controller/UserController.js
const axios = require("axios");
var userService = require("../service/UserService");

class UserController {
  async getUserBalance(req, res) {
    try {
      let balance = await userService.getUserBalance(req.params.userId);
      res.send(balance);
    } catch (err) {
      throw new Error(error.response.data.message);
    }
  }

  async getUser(req, res) {
    try {
      let user = await userService.getUser(req.params.userId);
      res.send(user);
    } catch (err) {
      throw new Error(error.response.data.message);
    }
  }
}

module.exports = new UserController();
